newIDE/app/src/UI/Theme/index.js

```diff
@@ -1,23 +1,61 @@
 // @flow
+import {
+  select,
+  blend as Install,
+} from '../../Utils/Object';
+import { isLTR } from '../../Utils/i18n/language';
+import domain from './domain';
+import DefaultTheme, {
+  themeName as defaultThemeName,
+  type Theme,
+} from './DefaultTheme';
 import DarkTheme from './DarkTheme';
-import DefaultTheme from './DefaultTheme';
 import NordTheme from './NordTheme';
 import SolarizedDarkTheme from './SolarizedDarkTheme';
-import { type Theme } from './DefaultTheme';
 import './Global.css';
 
+type MuiTheme = $Exact<typeof DefaultTheme.muiTheme>;
+
+/**
+ * Enhances every Theme instance with a `muiTheme` property accessor.
+ *
+ * This ensures the `muiTheme` is correctly applied whenever language
+ * directionality is changed. Thus, based on `language-ltr`, one of
+ * two _distinct_ `muiTheme` instances is returned; these two instances
+ * are exactly alike (i.e. have the exact same properties).
+ */
+const install = new Install(({ gdevelopTheme, muiTheme }: Theme): Theme => {
+  const muiThemes: [MuiTheme, MuiTheme] = [muiTheme, { ...muiTheme }];
+  return {
+    gdevelopTheme,
+    get muiTheme(): MuiTheme {
```

Comment:
I'm not a big fan of overriding accessors. It makes reasoning on the code harder as you have side effects that are not shown at the caller site. 
Let's see instead if we can do all the work in getTheme?